azure-proto-core/Resources/Sku.cs

```diff
@@ -17,12 +16,30 @@ public class Sku : IEquatable<Sku>, IComparable<Sku>
 
         public int CompareTo(Sku other)
         {
-            throw new NotImplementedException();
+            if (other == null) return 1;
+            if (this.Name == null && other.Name != null) return -1;
+            if (this.Name != null && this.Name.CompareTo(other.Name) != 0) return this.Name.CompareTo(other.Name);
```

Comment:
store this.Name.CompareTo(other.Name) so you don't have to call it twice

this applies to all the other items in this method and Equals